test/test_open.py

```diff
@@ -129,8 +140,15 @@ def test_file_perms(self, safer_open):
             fp.write('hello')
         assert FILENAME.read_text() == 'hello'
         mode = os.stat(FILENAME).st_mode
-        assert mode in (0o100664, 0o100644), stat.filemode(mode)
-        new_mode = mode & 0o100770
+
+        # UNIX systems view and manipulate file permissions as bits
+        if os.name == 'posix':
+            assert mode in (0o100664, 0o100644), stat.filemode(mode)
+            new_mode = mode & 0o100770
+        # disparity in Windows file handling
+        elif os.name == 'nt':
+            # instead, just check if its a regular file without permission bit specifics
+            new_mode = mode
```

Comment:
1. The AI had some sort of Windows-y test for modes here, I didn't look at it, you might try it...?
2. You need another `else` to catch the other cases, whatever they might be:

```
else:
    assert False, f'Do not understand os.name = {os.name}'
```

3. Should be `# Instead, just check if it's a regular file without permission bit specifics` - capital I, apostrophe.

Reply:
For:
1. I wanted a human opinion, first. 😅
2. Spot on! I'll get to that.
3. Thanks a bunch for that fix.

Reply:
Also, for the Windows-y mode, I'll have to go through the code in more detail and check docs, because a chunk of it was new to me. I'll give feedback in a couple of hours, though.